Advanced/Play Asset Delivery/Assets/PlayAssetDelivery/Runtime/InitializeAddressables.cs

```diff
@@ -0,0 +1,118 @@
+                if (coreUnityAssetPackNames.Length == 0)
+                    Debug.LogError("Cannot retrieve core Unity asset pack names. PlayCore Plugin is not installed.");
+                else
+                    AndroidAssetPacks.DownloadAssetPackAsync(coreUnityAssetPackNames, CheckDownloadStatus);
+            }
+#else
+            Setup();
+#endif
+        }
+
+        void Setup()
+        {
+            Addressables.ResourceManager.ResourceProviders.Add(new PlayAssetDeliveryAssetBundleProvider());
+            Addressables.ResourceManager.InternalIdTransformFunc = AssetPackTransformFunc;
+            hasInitialized = true;
+        }
+
+        void CheckDownloadStatus(AndroidAssetPackInfo info)
+        {
+            if (info.status == AndroidAssetPackStatus.Failed)
+                Debug.LogError($"Failed to retrieve the state of asset pack '{info.name}'."); 
+            else if (info.status == AndroidAssetPackStatus.Unknown)
+                Debug.LogError($"Asset pack '{info.name}' is unavailable for this application. This can occur if the app was not installed through Google Play.");
+            else if (info.status == AndroidAssetPackStatus.Canceled)
+                Debug.LogError($"Cancelled asset pack download request '{info.name}'.");
+            else if (info.status == AndroidAssetPackStatus.WaitingForWifi)
+            {
+                AndroidAssetPacks.RequestToUseMobileDataAsync(result =>
+                {
+                    if (!result.allowed)
+                        Debug.LogError("Request to use mobile data was denied.");
+                });
+            }
+            else if(info.status == AndroidAssetPackStatus.Completed)
+            {
+                string assetPackPath = AndroidAssetPacks.GetAssetPackPath(info.name);
+                if (string.IsNullOrEmpty(assetPackPath))
+                {
+                    Debug.LogError($"Downloaded asset pack '{info.name}' but cannot locate it on device.");
+                }
+            }
+        }
+    
+        string AssetPackTransformFunc(IResourceLocation location)
+        {
+
+#if UNITY_ANDROID && !UNITY_EDITOR
+            string filename = Path.GetFileName(location.InternalId);
+
+            if (location.ResourceType == typeof(IAssetBundleResource))
+            {
+                string assetPackName = Path.GetFileNameWithoutExtension(location.InternalId);
+                if (PlayerPrefs.HasKey(assetPackName))
+                {
+                    // Load bundle from a custom asset pack. PlayAssetDeliveryBundleProvider.Provider previously stored its path.
+                    return PlayerPrefs.GetString(assetPackName);
+                }
+            }
+
+            string streamingAssetsPackPath = AndroidAssetPacks.GetAssetPackPath("UnityStreamingAssetsPack");
+            if (!string.IsNullOrEmpty(streamingAssetsPackPath))
```

Comment:
The supported way of accessing streaming assets path is by using Application.streamingAssetsPath API ( https://docs.unity3d.com/Manual/StreamingAssets.html ) which will return the correct streaming assets path which can be fed to UnityWebRequest no matter where those streaming assets are actually placed (in base application, in data asset pack (either install-time or fast-follow delivered), in separate streaming assets asset pack (either install-time or fast-follow delivered).
I am not sure if this code is actually depending on the VFS behavior of fallbacking to correct "split" to load the asset bundle from the install-time delivered custom asset pack or if the code actually tries to get a path to the asset bundle in the streaming assets location, in which case Application.streamingAssetsPath is the correct path to use.